ledger/complete/ledger_test.go

```diff
@@ -355,6 +355,75 @@ func TestLedgerFunctionality(t *testing.T) {
 	}
 }
 
+func Test_ExportCheckpointAt(t *testing.T) {
+
+	t.Run("noop migration", func(t *testing.T) {
+		unittest.RunWithTempDir(t, func(dbDir string) {
+			unittest.RunWithTempDir(t, func(dir2 string) {
+
+				led, err := complete.NewLedger(dbDir, 100, &metrics.NoopCollector{}, zerolog.Logger{}, nil, complete.DefaultPathFinderVersion)
+				require.NoError(t, err)
+
+				state := led.InitialState()
+				u := utils.UpdateFixture()
+				u.SetState(state)
+
+				state, err = led.Set(u)
+				require.NoError(t, err)
+
+				newState, err := led.ExportCheckpointAt(state, []ledger.Migration{noOpMigration}, complete.DefaultPathFinderVersion, dir2+"/root.checkpoint")
+				require.NoError(t, err)
+				assert.Equal(t, newState, state)
+
+				led2, err := complete.NewLedger(dir2, 100, &metrics.NoopCollector{}, zerolog.Logger{}, nil, complete.DefaultPathFinderVersion)
+				require.NoError(t, err)
+
+				q, err := ledger.NewQuery(state, u.Keys())
+				require.NoError(t, err)
+
+				retValues, err := led2.Get(q)
+				require.NoError(t, err)
+
+				for i, v := range u.Values() {
+					assert.Equal(t, v, retValues[i])
+				}
+			})
+		})
+	})
+	t.Run("change migration", func(t *testing.T) {
+		unittest.RunWithTempDir(t, func(dbDir string) {
+			unittest.RunWithTempDir(t, func(dir2 string) {
+
+				led, err := complete.NewLedger(dbDir, 100, &metrics.NoopCollector{}, zerolog.Logger{}, nil, complete.DefaultPathFinderVersion)
+				require.NoError(t, err)
+
+				state := led.InitialState()
+				u := utils.UpdateFixture()
+				u.SetState(state)
+
+				state, err = led.Set(u)
+				require.NoError(t, err)
+
+				newState, err := led.ExportCheckpointAt(state, []ledger.Migration{changeOnesMigration}, complete.DefaultPathFinderVersion, dir2+"/root.checkpoint")
```

Comment:
A comment here mentioning that this function changes any value that's equal to "A", to "C" would make the test more clear

Reply:
Added comments at c3cc915